src/irr/asset/CGLIWriter.cpp

```diff
@@ -0,0 +1,327 @@
+
+			case EF_R32_UINT: return getTranslatedFinalFormat(FORMAT_R32_UINT_PACK32);				//GL_R32UI
+			case EF_R32G32_UINT: return getTranslatedFinalFormat(FORMAT_RG32_UINT_PACK32);			//GL_RG32UI
+			case EF_R32G32B32_UINT: return getTranslatedFinalFormat(FORMAT_RGB32_UINT_PACK32);			//GL_RGB32UI
+			case EF_R32G32B32A32_UINT: return getTranslatedFinalFormat(FORMAT_RGBA32_UINT_PACK32);			//GL_RGBA32UI
+
+			case EF_A2R10G10B10_UINT_PACK32: return getTranslatedFinalFormat(FORMAT_RGB10A2_UINT_PACK32);			//GL_RGB10_A2UI
+			case EF_A2R10G10B10_SINT_PACK32: return getTranslatedFinalFormat(FORMAT_RGB10A2_SINT_PACK32);	//GL_RGB10_A2I
+
+			// signed integer formats
+			case EF_R8_SINT: return getTranslatedFinalFormat(FORMAT_R8_SINT_PACK8);				//GL_R8I
+			case EF_R8G8_SINT: return getTranslatedFinalFormat(FORMAT_RG8_SINT_PACK8);				//GL_RG8I
+			case EF_R8G8B8_SINT: return getTranslatedFinalFormat(FORMAT_RGB8_SINT_PACK8);			//GL_RGB8I
+			case EF_R8G8B8A8_SINT: return getTranslatedFinalFormat(FORMAT_RGBA8_SINT_PACK8);			//GL_RGBA8I
+
+			case EF_R16_SINT: return getTranslatedFinalFormat(FORMAT_R16_SINT_PACK16);				//GL_R16I
+			case EF_R16G16_SINT: return getTranslatedFinalFormat(FORMAT_RG16_SINT_PACK16);			//GL_RG16I
+			case EF_R16G16B16_SINT: return getTranslatedFinalFormat(FORMAT_RGB16_SINT_PACK16);			//GL_RGB16I
+			case EF_R16G16B16A16_SINT: return getTranslatedFinalFormat(FORMAT_RGBA16_SINT_PACK16);			//GL_RGBA16I
+
+			case EF_R32_SINT: return getTranslatedFinalFormat(FORMAT_R32_SINT_PACK32);				//GL_R32I
+			case EF_R32G32_SINT: return getTranslatedFinalFormat(FORMAT_RG32_SINT_PACK32);			//GL_RG32I
+			case EF_R32G32B32_SINT: return getTranslatedFinalFormat(FORMAT_RGB32_SINT_PACK32);			//GL_RGB32I
+			case EF_R32G32B32A32_SINT: return getTranslatedFinalFormat(FORMAT_RGBA32_SINT_PACK32);			//GL_RGBA32I
+
+			// Floating formats
+			case EF_R16_SFLOAT: return getTranslatedFinalFormat(FORMAT_R16_SFLOAT_PACK16);				//GL_R16F
+			case EF_R16G16_SFLOAT: return getTranslatedFinalFormat(FORMAT_RG16_SFLOAT_PACK16);			//GL_RG16F
+			case EF_R16G16B16_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGB16_SFLOAT_PACK16);			//GL_RGB16F
+			case EF_R16G16B16A16_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGBA16_SFLOAT_PACK16);			//GL_RGBA16F
+
+			case EF_R32_SFLOAT: return getTranslatedFinalFormat(FORMAT_R32_SFLOAT_PACK32);				//GL_R32F
+			case EF_R32G32_SFLOAT: return getTranslatedFinalFormat(FORMAT_RG32_SFLOAT_PACK32);			//GL_RG32F
+			case EF_R32G32B32_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGB32_SFLOAT_PACK32);			//GL_RGB32F
+			case EF_R32G32B32A32_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGBA32_SFLOAT_PACK32);			//GL_RGBA32F
+
+			case EF_R64_SFLOAT: return getTranslatedFinalFormat(FORMAT_R64_SFLOAT_PACK64);			//GL_R64F
+			case EF_R64G64_SFLOAT: return getTranslatedFinalFormat(FORMAT_RG64_SFLOAT_PACK64);		//GL_RG64F
+			case EF_R64G64B64_SFLOAT: return getTranslatedFinalFormat(FORMAT_RGB64_SFLOAT_PACK64);		//GL_RGB64F
```

Comment:
formats which don't fit in Power-Of-Two for a texel/block will never be allowed to be texture formats (only vertex formats)